File: ruby/gmsec4/dox/ConfigFileIterator.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.


module Libgmsec_ruby #:nodoc: don't document this

	# This class is used to access elements of a ConfigFile object.
	#
	class ConfigFileIterator
		# Reports whether there is a next CONFIG entry available via the iterator.
		#
		# ==== Returns
		#
		# Returns true if another CONFIG entry is available, false otherwise.
		#
		def has_next_config
		end


		# Reports whether there is a next MESSAGE entry available via the iterator.
		#
		# ==== Returns
		#
		# Returns true if another MESSAGE entry is available, false otherwise.
		#
		def has_next_message
		end


		# Reports whether there is a next SUBSCRIPTION entry available via the iterator.
		#
		# ==== Returns
		#
		# Returns true if another SUBSCRIPTION entry is available, false otherwise.
		#
		def has_next_subscription
		end


		# Reports whether there is a next SUBSCRIPTION entry available via the iterator.
		#
		# ==== Returns
		#
		# Returns true if another SUBSCRIPTION entry is available, false otherwise.
		#
		def has_next_subscription
		end


		# Reports whether there is a next CUSTOM entry available via the iterator.
		#
		# ==== Returns
		#
		# Returns true if another CUSTOM entry is available, false otherwise.
		#
		def has_next_custom_element
		end


		##
		# Provides a reference to a ConfigEntry object that represents the next CONFIG entry.
		#
		# ==== Returns
		#
		# A reference to a ConfigEntry.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if no more CONFIG entries are available via the iterator.
		#
		# ==== See also
		#
		# ConfigFileIterator.has_next_config
		#
		def next_config
		end


		##
		# Provides a reference to a MessageEntry object that represents the next MESSAGE entry.
		#
		# ==== Returns
		#
		# A reference to a MessageEntry.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if no more MESSAGE entries are available via the iterator.
		#
		# ==== See also
		#
		# ConfigFileIterator.has_next_message
		#
		def next_message
		end


		##
		# Provides a reference to a SubscriptionEntry object that represents the next SUBSCRIPTION entry.
		#
		# ==== Returns
		#
		# A reference to a SubscriptionEntry.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if no more SUBSCRIPTION entries are available via the iterator.
		#
		# ==== See also
		#
		# ConfigFileIterator.has_next_subscription
		#
		def next_subscription
		end


		##
		# Returns an XML string representing the next available CUSTOM entry.
		#
		# ==== Returns
		#
		# An XML string.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if no more CUSTOM entries are available via the iterator.
		#
		# ==== See also
		#
		# ConfigFileIterator.has_next_custom_element
		#
		def next_custom_element
		end


		##
		# Resets all iterators to the beginning of the respective lists that are maintained by the ConfigFile.
		#
		def reset
		end
	end
end

File: csharp/gmsec4/dox/ConfigFileIterator.cs
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>
/// This class is used to access elements of a configuration file (ConfigFile) object.
/// </summary>

public class ConfigFileIterator : global::System.IDisposable {

  /// <summary>Reports back whether additional configuration is available</summary>
  /// <returns>True if an additional configuration is available; false otherwise.</returns>
  public bool HasNextConfig() {
  }

  /// <summary>Reports back whether additional message is available</summary>
  /// <returns>True if an additional message is available; false otherwise.</returns>
  public bool HasNextMessage() {
  }

  /// <summary>Reports back whether additional subscription is available</summary>
  /// <returns>True if an additional subscriptions is available; false otherwise.</returns>
  public bool HasNextSubscription() {
  }

  /// <summary>Reports back whether additional custom XML element is available</summary>
  /// <returns>True if an additional custom XML element is available; false otherwise.</returns>
  public bool HasNextCustomElement() {
  }

  /// <summary>Reports back the next available custom XML element.</summary>
  /// <returns>The custom XML element string.</returns>
  /// <exception cref="GmsecException">Thrown if there is not an additional custom XML element.</exception>
  public string NextCustomElement() {
  }

  /// <summary>Resets all iterators to the beginning of the respective lists that are maintained by the ConfigFile.</summary>
  public void Reset() {
  }

  /// <summary>Reports back the next available configuration.</summary>
  /// <returns>The next available ConfigEntry.</returns>
  /// <exception cref="GmsecException">Thrown if there is not an additional configuration entry.</exception>
  public ConfigEntry NextConfig() {
  }

  /// <summary>Reports back the next available message.</summary>
  /// <returns>The next available MessageEntry.</returns>
  /// <exception cref="GmsecException">Thrown if there is not an additional message entry.</exception>
  public MessageEntry NextMessage() {
  }

  /// <summary>Reports back the next available subscription.</summary>
  /// <returns>The next available SubscriptionEntry.</returns>
  /// <exception cref="GmsecException">Thrown if there is not an additional subscription entry.</exception>
  public SubscriptionEntry NextSubscription() {
  }

}

}
